docs/src/Combinatorics/partitions.md

```diff
@@ -0,0 +1,36 @@
+# Partitions
+
+```@docs
+    Partition
+```
+
+## Unrestricted partitions
+
+```@docs
+    partitions(::Integer)
```

Comment:
Meta-question: what is `Integer` here? The Julia type `Base.Integer`?

Can one also pass the OSCAR integer type `fmpz` to these functions?

Reply:
Yes, it's `Base.Integer`. Passing `fmpz` results in an error. Should we do something about this?

Reply:
It would be great if `IntegerUnion` was accepted, i.e. both `fmpz` (= `ZZRingElem`) and `Base.Integer`

Reply:
(But this is not a requirement for merging, just something that would be nice to have eventually -- if we don't do it in this PR, someone should open an issue to remind us to do it)

Reply:
I fixed `partitions` to accept `fmpz` integers. However, there are a number of other functions that take integers as inputs. Should we modify everything so that they all accept `IntegerUnion`?

Reply:
I changed all functions in `partitions.jl` to accept `IntegerUnion`. I didn't do that for `schur_polynomials.jl` or `tableaux.jl`. Hope this doesn't break anything... 😬

Reply:
> I fixed `partitions` to accept `fmpz` integers. However, there are a number of other functions that take integers as inputs. Should we modify everything so that they all accept `IntegerUnion`?

That was my problem as well and I didn't really know what to do about it.

Reply:
Ideally, any place that would accept a BigInt should also accept an `fmpz` / `ZZRingElem`.

Of course there are places where really only a "small" integer makes sense, in those case staying with `Integer` is fine.